# Customer Churn Analysis — Telecom Industry
## Notebook 1: Problem Definition & First Look

**Author:** Prakhar Sharma  
**Date:** 26 March 2026  
**Dataset:** IBM Telco Customer Churn (7,043 records)

---

## * Business Context
A telecom company is losing customers to competitors.
The goal is to identify WHO is likely to churn, WHY they churn,
and WHAT the business can do to retain them.

## * Business Questions We Will Answer
1. What is the overall churn rate?
2. Which customer segments churn the most?
3. What are the top drivers of churn?
4. Can we predict which customers will churn next month?
5. What retention strategies should the business adopt?

## * How We Define Churn
A customer is churned if they have discontinued the service
(Churn column = 'Yes' in the dataset)

In [2]:
# importing necessary libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
# importing dataset from drive
from google.colab import drive
drive.mount('/content/drive')
# creating  a dataframe from the filepath
df=pd.read_csv('/content/drive/MyDrive/projects /customer churn analysis/Data/raw/Telco-Customer-Churn.csv')
print("dataset loaded")
print(f"shape:{df.shape}")

Mounted at /content/drive
dataset loaded
shape:(7043, 21)


In [3]:
# to see the data looks , we see first five rows
df.head()

,customerID,gender,SeniorCitizen,Partner,Dependents,tenure,PhoneService,MultipleLines,InternetService,OnlineSecurity,...,DeviceProtection,TechSupport,StreamingTV,StreamingMovies,Contract,PaperlessBilling,PaymentMethod,MonthlyCharges,TotalCharges,Churn
0,7590-VHVEG,Female,0,Yes,No,1,No,No phone service,DSL,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,29.85,29.85,No
1,5575-GNVDE,Male,0,No,No,34,Yes,No,DSL,Yes,...,Yes,No,No,No,One year,No,Mailed check,56.95,1889.5,No
2,3668-QPYBK,Male,0,No,No,2,Yes,No,DSL,Yes,...,No,No,No,No,Month-to-month,Yes,Mailed check,53.85,108.15,Yes
3,7795-CFOCW,Male,0,No,No,45,No,No phone service,DSL,Yes,...,Yes,Yes,No,No,One year,No,Bank transfer (automatic),42.30,1840.75,No
4,9237-HQITU,Female,0,No,No,2,Yes,No,Fiber optic,No,...,No,No,No,No,Month-to-month,Yes,Electronic check,70.70,151.65,Yes


In [4]:
# next we see basic info like column names , datatypes and null counts
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 7043 entries, 0 to 7042
Data columns (total 21 columns):
 #   Column            Non-Null Count  Dtype  
---  ------            --------------  -----  
 0   customerID        7043 non-null   object 
 1   gender            7043 non-null   object 
 2   SeniorCitizen     7043 non-null   int64  
 3   Partner           7043 non-null   object 
 4   Dependents        7043 non-null   object 
 5   tenure            7043 non-null   int64  
 6   PhoneService      7043 non-null   object 
 7   MultipleLines     7043 non-null   object 
 8   InternetService   7043 non-null   object 
 9   OnlineSecurity    7043 non-null   object 
 10  OnlineBackup      7043 non-null   object 
 11  DeviceProtection  7043 non-null   object 
 12  TechSupport       7043 non-null   object 
 13  StreamingTV       7043 non-null   object 
 14  StreamingMovies   7043 non-null   object 
 15  Contract          7043 non-null   object 
 16  PaperlessBilling  7043 non-null   object 


In [6]:
# statistical summary
df.describe()


,SeniorCitizen,tenure,MonthlyCharges
count,7043.000000,7043.000000,7043.000000
mean,0.162147,32.371149,64.761692
std,0.368612,24.559481,30.090047
min,0.000000,0.000000,18.250000
25%,0.000000,9.000000,35.500000
50%,0.000000,29.000000,70.350000
75%,0.000000,55.000000,89.850000
max,1.000000,72.000000,118.750000


In [8]:
# Checking  churn distribution
churn_counts = df['Churn'].value_counts()
# checking churn rate
churn_rate = df['Churn'].value_counts(normalize=True) * 100

print("------CHURN DISTRIBUTION------")
print(churn_counts)
print()
print("------CHURN RATE------")
print(churn_rate.round(2))

------CHURN DISTRIBUTION------
Churn
No     5174
Yes    1869
Name: count, dtype: int64

------CHURN RATE------
Churn
No     73.46
Yes    26.54
Name: proportion, dtype: float64


##  Phase 1 — Key Findings

- Dataset: 7,043 customers, 21 columns
- Churn Rate: **26.54%** (1,869 customers left)
- Industry average is ~15–20% → this company is underperforming
- **Class imbalance noticed**: 73% No vs 26% Yes → accuracy alone won't work in modeling
- TotalCharges column needs fixing (object type, should be numeric)

--> Next: Deep EDA to find WHY customers are churning